controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```diff
@@ -431,18 +436,20 @@ public SubsystemRegistration registerSubsystem(String name, int majorVersion, in
 
         @Override
         public SubsystemRegistration registerSubsystem(String name, int majorVersion, int minorVersion, int microVersion, boolean deprecated) {
+            return registerSubsystem(name, ModelVersion.create(majorVersion, minorVersion, microVersion), deprecated);
+        }
+
+        public SubsystemRegistration registerSubsystem(String name, ModelVersion version, boolean deprecated) {
             assert name != null : "name is null";
             checkNewSubystem(extension.extensionModuleName, name);
             SubsystemInformationImpl info = extension.getSubsystemInfo(name);
-            info.setMajorVersion(majorVersion);
-            info.setMinorVersion(minorVersion);
-            info.setMicroVersion(microVersion);
+            info.setVersion(version);
```

Comment:
All the exisiting registerSubsystem() methods take one or more int parameters which cannot be null, so you end up with a version whether you like it or not